functions/src/bills/updateBillReferences.ts

```diff
@@ -92,53 +91,72 @@ class UpdateBillReferences extends BillProcessor {
   async getEventUpdates(): Promise<BillUpdates> {
     const hearings = await db
       .collection(`/events`)
-      .where("startsAt", ">=", new Date())
       .where("type", "==", "hearing")
       .get()
       .then(this.load(Hearing))
     const updates: BillUpdates = new Map()
 
-    // Set the next hearing on every bill referenced by upcoming hearings.
-    const billEvents = flattenDeep<{
-      startsAt: Timestamp
-      billId: string
-      hearingId: string
-      court: number
-    }>(
-      hearings.map(hearing =>
-        hearing.content.HearingAgendas.map(agenda =>
-          agenda.DocumentsInAgenda.map(doc => ({
-            startsAt: parseApiDateTime(agenda.StartTime),
-            billId: doc.BillNumber,
-            court: doc.GeneralCourtNumber,
-            hearingId: hearing.id
-          }))
-        )
-      )
-    )
-    billEvents.forEach(event => {
-      const existing = updates.get(event.billId)
-      if (!existing || event.startsAt < (existing.nextHearingAt as Timestamp)) {
-        updates.set(event.billId, {
-          nextHearingAt: event.startsAt,
-          nextHearingId: event.hearingId
+    // Build mapping from billId -> hearingIds and compute earliest upcoming hearing
+    const hearingIdsByBill = new Map<string, Set<string>>()
+
+    const now = Timestamp.fromMillis(Date.now())
+
+    hearings.forEach(hearing => {
+      const hearingId = hearing.id
```

Comment:
nit: I think this would be more convenient for later consumption if the `hearingIds` use `hearing.content.EventId` instead - for a hearing event, the `hearing.id` property will be something like `hearing-1234`, while `hearing.content.EventId` would be the pure id part of `1234`. Given that we can always assume that these ids are for Hearings, that would save clients from needing to parse out the only part they care.